modules/basics/src/main/java/com/opengamma/strata/basics/currency/CurrencyAmount.java

```diff
@@ -117,14 +115,19 @@ public static CurrencyAmount of(String currencyCode, double amount) {
    */
   @FromString
   public static CurrencyAmount parse(String amountStr) {
-    ArgChecker.notNull(amountStr, "amountStr");
-    List<String> split = Splitter.on(' ').splitToList(amountStr);
-    if (split.size() != 2) {
+    if (amountStr == null || amountStr.length() <= 4 || amountStr.charAt(3) != ' ') {
```

Comment:
Add a comment to indicate that the method has been optimised, eg `// this method has had some performance optimisations applied`